### PR TITLE
fix(wms): undefined fields can be set to 'undefined' strings

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: modules/wms/src/index.ts [hunk 1]
```diff
@@ -17,9 +17,13 @@ export {CSWRecordsLoader} from './csw-records-loader';
 
 export {WMSErrorLoader} from './wms-error-loader';
 
-export type {WMSLoaderOptions} from './wms-capabilities-loader';
-
-export type {WMSCapabilities} from './wms-capabilities-loader';
+export type {
+  WMSCapabilities,
```
Comment (ibgreen) [inline modules/wms/src/index.ts:21]: Exporting the missing types

File: modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts [hunk 2]
```diff
@@ -143,7 +143,7 @@ export type WMSExceptions = {
   mimeTypes: string[];
 };
 
-export type parseWMSCapabilitiesOptions = {
+export type ParseWMSCapabilitiesOptions = {
   /** Add inherited layer information to sub layers */
   inheritedLayerProps?: boolean;
```
Comment (ibgreen) [inline modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts:148]: By setting this prop, loaders.gl will inject inherited props on all child layers. It does make the resulting JSON massive.

File: modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts [hunk 3]
```diff
@@ -254,8 +253,8 @@ function extractLayer(xmlLayer: any): WMSLayer {
     // All layers must have a title
     title: String(xmlLayer?.Title || ''),
     // Name is required only if renderable
-    name: String(xmlLayer?.Name),
-    abstract: String(xmlLayer?.Abstract),
+    name: xmlLayer?.Name && String(xmlLayer?.Name),
```
Comment (ibgreen) [inline modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts:256]: Fix for 'undefined'

File: modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts [hunk 4]
```diff
@@ -406,19 +405,21 @@ function extractDimension(xmlDimension: any): WMSDimension {
...
     layer.crs = [...parent.crs];
   }
 
-  if (parent && parent.boundingBoxes && !layer.boundingBoxes) {
+  if (parent?.boundingBoxes && !layer.boundingBoxes) {
     layer.boundingBoxes = [...parent.boundingBoxes];
   }
 
-  // TODO inherit other elements
+  if (parent?.dimensions && !layer.dimensions) {
```
Comment (ibgreen) [inline modules/wms/src/lib/parsers/wms/parse-wms-capabilities.ts:420]: Add dimensions if inherited from parent

**Review by jwasilgeo — APPROVED** [review]

**Pull request merged**